### PR TITLE
A bunch of fixes

### PR DESCRIPTION
**Pull request by purcell**
Hi! I came across langserver recently, and looked at this code in relation to it. I administer MELPA, and we've got a different version of lsp-mode packaged there right now. I'd like to replace it with this version, and noticed a few details in the code which would benefit from being cleaned up. This PR includes a number of small fixes to that end: they should all be safe, but please double-check, because I haven't had the opportunity to regression test them against the old code.

When merged, I'll see about pointing MELPA's lsp-mode recipe at this repo instead of the other one.

Cheers!

### CODE REVIEW

**Comment by sqs** [comment]
Thank you! @beyang or I will review shortly.

File: lsp-mode.el [hunk 1]
```diff
@@ -256,16 +269,17 @@
 ;;; recursive call to init-lsp-conn-inner, although it's a bit challenging due
 ;;; to the callback nature.
 
-(defun init-lsp-conn-inner (ws conn)
+(defun lsp-init-conn-inner (ws conn)
   (send-lsp-msg (lsp-init (lsp-init-params nil nil ws nil))
```
Comment (beyang) [inline lsp-mode.el]: s/send-lsp-msg/lsp-send-msg/

File: lsp-mode.el [hunk 2]
```diff
@@ -274,46 +288,50 @@
...
          (ws (projectile-project-root))
-         (conn (make-ls-connection :process net-proc :session session :bufname bufname))
+         (conn (make-lsp-connection :process net-proc :session session :bufname bufname))
          )
     (set-process-filter net-proc (lsp-filter session))
     (puthash ws conn lsp-ws-connection-map)
-    (init-lsp-conn-inner ws conn)
-    (send-lsp-msg (lsp-did-open-text-doc (lsp-text-doc-id (uri-for-path (buffer-file-name)))) 'lsp-ignore)
+    (lsp-init-conn-inner ws conn)
+    (send-lsp-msg (lsp-did-open-text-doc (lsp-buffer-text-doc-id)) 'lsp-ignore)
```
Comment (beyang) [inline lsp-mode.el]: s/send-lsp-msg/lsp-send-msg/

**Review by beyang — CHANGES_REQUESTED** [review]
Hi there, thanks for submitting this PR. I'm currently trying to test this against the Go language server (https://github.com/sourcegraph/go-langserver), but am encountering silent failure on all the LSP methods. Initially, I saw a "Symbol's definition is void" error, which was fixed by the changes in the inline comments, but there appears to be at least one more issue. Can you share which LSP implementation you're testing against?

File: lsp-mode.el (hunk 1 above)
Comment (purcell) [reply lsp-mode.el]: Yep, those are fixed now, sorry: `highlight-symbol-query-replace` had failed to replace all occurrences. I've pushed fresh changes.

**Comment by purcell** [comment]
I have the go-langserver installed locally, together with the repo containing its code. In which directory should I start the server, and what would be a reasonable test? I've tried starting it in the root dir of its own repo, with the trace option enabled, and then poking at it with `lsp-mode` from a buffer visiting (say) `ast.go`, but I get mostly nonsensical errors, e.g.
```
--> request #1397201556940209679: textDocument/references: {"context":{"includeDeclaration":false},"textDocument":{"uri":"file:///tmp/go-langserver/langserver/ast.go"},"position":{"line":10,"character":7}}
<-- error #1397201556940209679: textDocument/references: {"code":0,"message":"import \"/tmp/go-langserver/langserver\": cannot import absolute path","data":null}
```

**Comment by beyang** [comment]
I uncovered a bug in the Go language server that was causing `lsp-mode-symbol` requests to fail (https://github.com/sourcegraph/go-langserver/issues/176). If I comment out this line in the plugin (https://github.com/sourcegraph/emacs-lsp/commit/041e092ab168d4097962cf9bbff566ee84d2cf17), it works against the Go language server. This is a bug on the Go language server side, so no need to change this PR.

**Review by beyang — APPROVED** [review]

**Comment by beyang** [comment]
Thanks for the patch!

**Comment by purcell** [comment]
Cool, happy to help. Having also reviewed this project's namesake, https://github.com/vibhavp/emacs-lsp, I'd say that other version is quite a bit more complete, and you might consider joining forces to make that one fit your needs too.

**Pull request merged**